### DISCUSSION TITLE
Failed to get VBIOS from VRAM and kernel panic during early boot

### DISCUSSION THREAD

**Discussion by LatifSC** [ChefKissInc/NootedRed#26] [General]
Ryzen 3750H with Vega 10 and 1660ti
I saw a screenshot of Vega 10 running macos with this plugin. I gave it a try but doesnt work for me. Am I doing something wrong or it just doesnt support my system?

macOS 11.7.1
Lilu 1.6.4
Opencore 0.9.0

**Comment by VisualEhrmanntraut** [comment]
Ensure Legacy Boot/CSM is disabled and SMBIOS selection is as described in the README

**Reply by LatifSC** [reply]
I did. Now it boots to black screen. And macos is working. No panic/crash.

**Reply by VisualEhrmanntraut** [reply]
Are you using SMBIOS MacBookPro16,3 (laptop) or MacPro7,1 (desktop)? Also, are you using WhateverGreen? If so, it needs to be removed.

**Reply by LatifSC** [reply]
WhateverGreen is removed. Its still the same. I'm using MacBookPro16,4

**Reply by jalavoui** [reply]
The ATY,bin_image is not properly injected by the driver - it should be "visible" in IOService tab using IORegistryExplorer.  Current release inject the bios from VFCT table just fine. Issue is on other AMD drivers the property is "visible" because it's done inside some driver function.

Figured out why: in the past the code was done in __ZN14AmdTtlServicesC2EP11IOPCIDevice (the ATY,bin_image  property was visible in IORegistryExplorer IOService Tree)

**Reply by VisualEhrmanntraut** [reply]
> The ATY,bin_image is not properly injected by the driver - it should be "visible" in IOService tab using IORegistryExplorer.

Can you elaborate?

**Reply by VisualEhrmanntraut** [reply]
> The ATY,bin_image is not properly injected by the driver - it should be "visible" in IOService tab using IORegistryExplorer. Current release inject the bios from VFCT table just fine. Issue is on other AMD drivers the property is "visible" because it's done inside some driver function.
> 
> Figured out why: in the past the code was done in __ZN14AmdTtlServicesC2EP11IOPCIDevice (the ATY,bin_image property was visible in IORegistryExplorer IOService Tree)

AMDRadeonX6000FB removes it. Our code is fine
